nilearn/_utils/cache_mixin.py

```diff
@@ -226,19 +237,9 @@ def _cache(self, func, func_memory_level=1, shelve=False, **kwargs):
             For consistency, a callable object is always returned.
 
         """
-        verbose = getattr(self, "verbose", 0)
-
-        # Creates attributes if they don't exist
-        # This is to make creating them in __init__() optional.
-        if not hasattr(self, "memory_level"):
-            self.memory_level = 0
-        if not hasattr(self, "memory"):
-            self.memory = Memory(location=None, verbose=verbose)
-        self.memory = check_memory(self.memory, verbose=verbose)
-
         # If cache level is 0 but a memory object has been provided, set
         # memory_level to 1 with a warning.
-        if self.memory_level == 0 and self.memory.location is not None:
+        if self.memory_level == 0 and self.memory_.location is not None:
```

Comment:
It might be good to also check `self.memory_` is not `None`.

Reply:
ok to be extra safe I am going to modify the way each estimator that uses cache_mixin checks it is fitted to include something to make sure memory_ has been set properly